src/Offer/Offer.php

```diff
@@ -156,7 +176,39 @@ public function updateTheme(Theme $theme)
      */
     public function updateFacilities(array $facilities)
     {
-        $this->apply($this->createFacilitiesUpdatedEvent($facilities));
+        if (empty($this->facilities) || !$this->sameFacilities($this->facilities, $facilities)) {
+            $this->apply($this->createFacilitiesUpdatedEvent($facilities));
+        }
+    }
+
+    /**
+     * @param AbstractFacilitiesUpdated $facilitiesUpdated
+     */
+    protected function applyFacilitiesUpdated(AbstractFacilitiesUpdated $facilitiesUpdated)
+    {
+        $this->facilities = $facilitiesUpdated->getFacilities();
+    }
+
+    /**
+     * @param array $facilities1
+     * @param array $facilities2
+     * @return bool
+     */
+    private function sameFacilities($facilities1, $facilities2)
+    {
+        if (count($facilities1) !== count($facilities2)) {
+            return false;
+        }
+
+        $sameFacilities = array_uintersect(
```

Comment:
Zou het niet logischer zijn om `array_udiff()` te gebruiken aangezien we willen checken of er verschillen zijn of niet?

Reply:
Ik twijfel of we dan wel dan wel nieuwe facilities meepakken. http://php.net/manual/en/function.array-udiff.php
> Returns an array containing all the values of array1 that are not present in any of the other arguments.

Reply:
Je hebt gelijk, dan zou je 2 diffs moeten maken (1 voor ontbrekende facilities, en 1 voor toegevoegde facilities), en als één of beiden niet leeg zijn dan zijn er toegevoegd/verwijderd. Maar dat zal minder performant zijn gok ik dus laat het maar zo met de intersect. Creatieve oplossing wel. 👍